some warnings

1.
```
In file included from /home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce_test.cc:17:0:
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce_test.cc: In member function 'virtual void ENFORCE_GT_FAIL_Test::TestBody()':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:302:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, >, <=, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce_test.cc:121:5: note: in expansion of macro 'PADDLE_ENFORCE_GT'
     PADDLE_ENFORCE_GT(1, 2UL);
     ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce_test.cc: In member function 'virtual void ENFORCE_GE_OK_Test::TestBody()':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:304:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, >=, <, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce_test.cc:132:3: note: in expansion of macro 'PADDLE_ENFORCE_GE'
   PADDLE_ENFORCE_GE(2, 2UL);
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:304:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, >=, <, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce_test.cc:133:3: note: in expansion of macro 'PADDLE_ENFORCE_GE'
   PADDLE_ENFORCE_GE(3, 2UL);
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce_test.cc: In member function 'virtual void ENFORCE_GE_FAIL_Test::TestBody()':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:304:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, >=, <, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce_test.cc:140:5: note: in expansion of macro 'PADDLE_ENFORCE_GE'
     PADDLE_ENFORCE_GE(1, 2UL);
```
2.
```
In file included from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/array.h:19:0,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/dim.h:22,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/ddim.h:20,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/lod_tensor.h:27,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/inference/api/api.cc:17:
/home/tangjian/paddle-tj-docker/paddle/fluid/inference/api/api.cc: In member function 'void paddle::PaddleBuf::Free()':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:302:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, >, <=, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/inference/api/api.cc:95:5: note: in expansion of macro 'PADDLE_ENFORCE_GT'
     PADDLE_ENFORCE_GT(length_, 0);
     ^
```
3. 
```
[13:51:45]/paddle/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h: In instantiation of 'void paddle::operators::EmbeddingVSumFunctor<T>::operator()(const paddle::framework::ExecutionContext&, const LoDTensor*, const LoDTensor*, paddle::operators::LoDTensor*) [with T = float; paddle::operators::LoDTensor = paddle::framework::LoDTensor]':
[13:51:45]/paddle/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h:74:14:   required from 'void paddle::operators::FusedEmbeddingSeqPoolKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = float]'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:107:11:   required from 'paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const::<lambda(const paddle::framework::ExecutionContext&)> [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 0ul; KernelTypes = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:106:10:   required from 'struct paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 0ul; KernelTypes = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]::<lambda(const class paddle::framework::ExecutionContext&)>'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:103:38:   required from 'void paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 0ul; KernelTypes = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:130:9:   required from 'paddle::framework::OpKernelRegistrar<PlaceType, KernelType>::OpKernelRegistrar(const char*, const char*, int) [with PlaceType = paddle::platform::CPUPlace; KernelType = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]'
[13:51:45]/paddle/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.cc:189:695:   required from here
[13:51:45]/paddle/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h:53:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[13:51:45]     for (int64_t i = 0; i != ids_lod.size() - 1; ++i) {
[13:51:45]                           ^
[13:51:45]/paddle/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h: In instantiation of 'void paddle::operators::EmbeddingVSumFunctor<T>::operator()(const paddle::framework::ExecutionContext&, const LoDTensor*, const LoDTensor*, paddle::operators::LoDTensor*) [with T = double; paddle::operators::LoDTensor = paddle::framework::LoDTensor]':
[13:51:45]/paddle/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h:74:14:   required from 'void paddle::operators::FusedEmbeddingSeqPoolKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = double]'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:107:11:   required from 'paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const::<lambda(const paddle::framework::ExecutionContext&)> [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 1ul; KernelTypes = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:106:10:   required from 'struct paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 1ul; KernelTypes = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]::<lambda(const class paddle::framework::ExecutionContext&)>'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:103:38:   required from 'void paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 1ul; KernelTypes = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:112:9:   required from 'void paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 0ul; KernelTypes = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]'
[13:51:45]/paddle/paddle/fluid/framework/op_registry.h:130:9:   required from 'paddle::framework::OpKernelRegistrar<PlaceType, KernelType>::OpKernelRegistrar(const char*, const char*, int) [with PlaceType = paddle::platform::CPUPlace; KernelType = {paddle::operators::FusedEmbeddingSeqPoolKernel<float>, paddle::operators::FusedEmbeddingSeqPoolKernel<double>}]'
[13:51:45]/paddle/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.cc:189:695:   required from here
[13:51:45]/paddle/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h:53:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```

4
```
from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/attribute.h:25:0,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/op_desc.h:20,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/node.h:21,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/details/var_handle.h:23,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/details/op_handle_base.h:19,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/details/data_balance_op_handle.h:19,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/details/data_balance_op_handle.cc:15:
/home/tangjian/paddle-tj-docker/paddle/fluid/framework/details/data_balance_op_handle.cc: In member function 'virtual void paddle::framework::details::DataBalanceOpHandle::RunImpl()':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:302:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, >, <=, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/framework/details/data_balance_op_handle.cc:89:3: note: in expansion of macro 'PADDLE_ENFORCE_GT'
   PADDLE_ENFORCE_GT(places_.size(), 1,
```

5.
```
 /home/tangjian/paddle-tj-docker/paddle/fluid/framework/details/fuse_vars_op_handle.cc: In member function 'virtual void paddle::framework::details::FuseVarsOpHandle::RunImpl()':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:298:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/framework/details/fuse_vars_op_handle.cc:26:3: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
   PADDLE_ENFORCE_EQ(in_var_handles.size(), 0);
```
6.
```
n file included from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/attribute.h:25:0,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/op_desc.h:20,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/node.h:21,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/graph.h:22,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/graph_helper.h:21,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/graph_pattern_detector.cc:20:
/home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/graph_pattern_detector.cc: In member function 'paddle::framework::ir::PDNode* paddle::framework::ir::PDPattern::NewNode(const string&)':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:298:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/graph_pattern_detector.cc:41:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
     PADDLE_ENFORCE_EQ(node_map_.count(name), 0,
     ^
/home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/graph_pattern_detector.cc: In member function 'paddle::framework::ir::PDNode* paddle::framework::ir::PDPattern::NewNode(paddle::framework::ir::PDNode::teller_t&&, const string&)':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:298:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/framework/ir/graph_pattern_detector.cc:54:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
     PADDLE_ENFORCE_EQ(node_map_.count(name), 0,
```
7.
```
In file included from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/data_layout.h:21:0,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/tensor.h:22,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/framework/eigen.h:17,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/operators/crf_decoding_op.h:17,
                 from /home/tangjian/paddle-tj-docker/paddle/fluid/operators/crf_decoding_op.cc:15:
/home/tangjian/paddle-tj-docker/paddle/fluid/operators/crf_decoding_op.cc: In member function 'virtual void paddle::operators::CRFDecodingOp::InferShape(paddle::framework::InferShapeContext*) const':
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:298:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/operators/crf_decoding_op.cc:87:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
     PADDLE_ENFORCE_EQ(emission_dims.size(), 2UL,
     ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:287:46: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
     if (UNLIKELY(!((__cond1__)__CMP(__cond2__)))) {                     \
                                              ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:129:64: note: in definition of macro 'UNLIKELY'
 #define UNLIKELY(condition) __builtin_expect(static_cast<bool>(condition), 0)
                                                                ^
/home/tangjian/paddle-tj-docker/paddle/fluid/platform/enforce.h:298:3: note: in expansion of macro '__PADDLE_BINARY_COMPARE'
   __PADDLE_BINARY_COMPARE(__VAL0, __VAL1, ==, !=, __VA_ARGS__)
   ^
/home/tangjian/paddle-tj-docker/paddle/fluid/operators/crf_decoding_op.cc:92:5: note: in expansion of macro 'PADDLE_ENFORCE_EQ'
     PADDLE_ENFORCE_EQ(transition_dims.size(), 2UL,
```

...

about 30 warnings

How about fix following warnings at the same time?
4. 
```
[14:03:04]/paddle/paddle/fluid/operators/attention_lstm_op.cc: In instantiation of 'void paddle::operators::AttentionLSTMKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = float]':
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:107:11:   required from 'paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const::<lambda(const paddle::framework::ExecutionContext&)> [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 0ul; KernelTypes = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]'
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:106:10:   required from 'struct paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 0ul; KernelTypes = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]::<lambda(const class paddle::framework::ExecutionContext&)>'
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:103:38:   required from 'void paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 0ul; KernelTypes = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]'
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:130:9:   required from 'paddle::framework::OpKernelRegistrar<PlaceType, KernelType>::OpKernelRegistrar(const char*, const char*, int) [with PlaceType = paddle::platform::CPUPlace; KernelType = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]'
[14:03:04]/paddle/paddle/fluid/operators/attention_lstm_op.cc:417:629:   required from here
[14:03:04]/paddle/paddle/fluid/operators/attention_lstm_op.cc:296:117: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[14:03:04]/paddle/paddle/fluid/operators/attention_lstm_op.cc: In instantiation of 'void paddle::operators::AttentionLSTMKernel<T>::Compute(const paddle::framework::ExecutionContext&) const [with T = double]':
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:107:11:   required from 'paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const::<lambda(const paddle::framework::ExecutionContext&)> [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 1ul; KernelTypes = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]'
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:106:10:   required from 'struct paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 1ul; KernelTypes = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]::<lambda(const class paddle::framework::ExecutionContext&)>'
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:103:38:   required from 'void paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 1ul; KernelTypes = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]'
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:112:9:   required from 'void paddle::framework::OpKernelRegistrarFunctor<PlaceType, false, I, KernelTypes ...>::operator()(const char*, const char*, int) const [with PlaceType = paddle::platform::CPUPlace; long unsigned int I = 0ul; KernelTypes = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]'
[14:03:04]/paddle/paddle/fluid/framework/op_registry.h:130:9:   required from 'paddle::framework::OpKernelRegistrar<PlaceType, KernelType>::OpKernelRegistrar(const char*, const char*, int) [with PlaceType = paddle::platform::CPUPlace; KernelType = {paddle::operators::AttentionLSTMKernel<float>, paddle::operators::AttentionLSTMKernel<double>}]'
[14:03:04]/paddle/paddle/fluid/operators/attention_lstm_op.cc:417:629:   required from here
[14:03:04]/paddle/paddle/fluid/operators/attention_lstm_op.cc:296:117: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```
5.
```
[14:03:05]/paddle/paddle/fluid/operators/fused/fusion_repeated_fc_relu_op.cc: In member function 'virtual void paddle::operators::FusionRepeatedFCReluOp::InferShape(paddle::framework::InferShapeContext*) const':
[14:03:05]/paddle/paddle/fluid/operators/fused/fusion_repeated_fc_relu_op.cc:40:118: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[14:03:05][ 35%] Built target decayed_adagrad_op
[14:03:05]/paddle/paddle/fluid/operators/fused/fusion_repeated_fc_relu_op.cc:52:123: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```
6. 
```
[14:03:05]/paddle/paddle/fluid/operators/fused/fusion_squared_mat_sub_op.cc: In member function 'virtual void paddle::operators::FusionSquaredMatSubOp::InferShape(paddle::framework::InferShapeContext*) const':
[14:03:05]/paddle/paddle/fluid/operators/fused/fusion_squared_mat_sub_op.cc:45:118: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```
7.
```
[14:03:05]/paddle/paddle/fluid/operators/fused/fusion_seqpool_concat_op.cc: In member function 'virtual void paddle::operators::FusionSeqPoolConcatOp::InferShape(paddle::framework::InferShapeContext*) const':
[14:03:05]/paddle/paddle/fluid/operators/fused/fusion_seqpool_concat_op.cc:42:123: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
```

--- a/paddle/fluid/framework/details/broadcast_op_handle.cc
+++ b/paddle/fluid/framework/details/broadcast_op_handle.cc
@@ -30,7 +30,7 @@ void BroadcastOpHandle::RunImpl() {
   VarHandle *in_var_handle;
   {
     auto in_var_handles = DynamicCast<VarHandle>(inputs_);
-    PADDLE_ENFORCE_EQ(in_var_handles.size(), 1,
+    PADDLE_ENFORCE_EQ(in_var_handles.size(), 1UL,
                       "The number of input should be one.");
     in_var_handle = in_var_handles[0];
   }

--- a/paddle/fluid/framework/details/data_balance_op_handle.cc
+++ b/paddle/fluid/framework/details/data_balance_op_handle.cc
@@ -86,7 +86,7 @@ std::vector<std::array<int, 3>> DataBalanceOpHandle::GetBalancePlan(
 }
 
 void DataBalanceOpHandle::RunImpl() {
-  PADDLE_ENFORCE_GT(places_.size(), 1,
+  PADDLE_ENFORCE_GT(places_.size(), 1UL,
                     "Data balance can only be enabled when the number of "
                     "places to run larger than 1.");
   auto in_var_handles = DynamicCast<VarHandle>(this->Inputs());

--- a/paddle/fluid/framework/details/fuse_vars_op_handle.cc
+++ b/paddle/fluid/framework/details/fuse_vars_op_handle.cc
@@ -23,7 +23,7 @@ void FuseVarsOpHandle::RunImpl() {
 
   auto in_var_handles = DynamicCast<VarHandle>(this->Inputs());
   auto out_var_handles = DynamicCast<VarHandle>(this->Outputs());
-  PADDLE_ENFORCE_EQ(in_var_handles.size(), 0);
+  PADDLE_ENFORCE_EQ(in_var_handles.size(), 0UL);
   PADDLE_ENFORCE_EQ(out_var_handles.size() - 1, inputs_numel_.size(), "");
 
   auto scope = local_scope_->FindVar(kLocalExecScopeName)->Get<Scope *>();

--- a/paddle/fluid/framework/details/reduce_op_handle.cc
+++ b/paddle/fluid/framework/details/reduce_op_handle.cc
@@ -153,7 +153,7 @@ void ReduceOpHandle::RunImpl() {
   {
     auto out_var_handles = DynamicCast<VarHandle>(outputs_);
 
-    PADDLE_ENFORCE_EQ(out_var_handles.size(), 1,
+    PADDLE_ENFORCE_EQ(out_var_handles.size(), 1UL,
                       "The number of output should be one.");
     out_var_handle = out_var_handles.front();
   }

--- a/paddle/fluid/framework/ir/conv_bn_fuse_pass.cc
+++ b/paddle/fluid/framework/ir/conv_bn_fuse_pass.cc
@@ -169,7 +169,7 @@ std::unique_ptr<ir::Graph> ConvBNFusePass::ApplyImpl(
       if (has_bias && conv->Op()->Input("Bias").size() > 0) {
         // reuse existing conv bias node
         auto conv_bias_names = conv->Op()->Input("Bias");
-        PADDLE_ENFORCE_EQ(conv_bias_names.size(), 1);
+        PADDLE_ENFORCE_EQ(conv_bias_names.size(), 1UL);
         auto* conv_bias_var = scope->FindVar(conv_bias_names[0]);
         auto* conv_bias_tensor = conv_bias_var->GetMutable<LoDTensor>();
         PADDLE_ENFORCE_EQ(conv_bias_tensor->dims(),

--- a/paddle/fluid/framework/ir/fuse_relu_depthwise_conv_pass.cc
+++ b/paddle/fluid/framework/ir/fuse_relu_depthwise_conv_pass.cc
@@ -111,7 +111,7 @@ std::unique_ptr<ir::Graph> FuseReluDepthwiseConvPass::FuseReluDepthwiseConv(
       xg_var = subgraph.at(xg)->Var();
     }
 
-    PADDLE_ENFORCE_EQ(layer_op->Input("Input").size(), 1);
+    PADDLE_ENFORCE_EQ(layer_op->Input("Input").size(), 1UL);
     PADDLE_ENFORCE_EQ(layer_op->Input("Input")[0], y_var->Name());
     layer_op->SetInput("Input", {x_var->Name()});
     subgraph.at(layer)->inputs.push_back(subgraph.at(x));
@@ -119,13 +119,13 @@ std::unique_ptr<ir::Graph> FuseReluDepthwiseConvPass::FuseReluDepthwiseConv(
     VLOG(4) << "replace " << y_var->Name() << " -> " << x_var->Name();
 
     if (!only_forward) {
-      PADDLE_ENFORCE_EQ(layer_g_op->Input("Input").size(), 1);
+      PADDLE_ENFORCE_EQ(layer_g_op->Input("Input").size(), 1UL);
       PADDLE_ENFORCE_EQ(layer_g_op->Input("Input")[0], y_var->Name());
       layer_g_op->SetInput("Input", {x_var->Name()});
       subgraph.at(layer_g)->inputs.push_back(subgraph.at(x));
       subgraph.at(x)->outputs.push_back(subgraph.at(layer_g));
 
-      PADDLE_ENFORCE_EQ(layer_g_op->Output(GradVarName("Input")).size(), 1);
+      PADDLE_ENFORCE_EQ(layer_g_op->Output(GradVarName("Input")).size(), 1UL);
       PADDLE_ENFORCE_EQ(layer_g_op->Output(GradVarName("Input"))[0],
                         yg_var->Name());
       layer_g_op->SetOutput(GradVarName("Input"), {xg_var->Name()});

--- a/paddle/fluid/framework/ir/graph_pattern_detector.cc
+++ b/paddle/fluid/framework/ir/graph_pattern_detector.cc
@@ -38,7 +38,7 @@ size_t PDPattern::id_ = 0UL;
 
 PDNode *PDPattern::NewNode(const std::string &name) {
   if (!name.empty()) {
-    PADDLE_ENFORCE_EQ(node_map_.count(name), 0,
+    PADDLE_ENFORCE_EQ(node_map_.count(name), 0UL,
                       "PDNode's name should be unique, get duplicate [%s]",
                       name);
   }
@@ -51,7 +51,7 @@ PDNode *PDPattern::NewNode(const std::string &name) {
 
 PDNode *PDPattern::NewNode(PDNode::teller_t &&teller, const std::string &name) {
   if (!name.empty()) {
-    PADDLE_ENFORCE_EQ(node_map_.count(name), 0,
+    PADDLE_ENFORCE_EQ(node_map_.count(name), 0UL,
                       "PDNode's name should be unique, get duplicate [%s]",
                       name);
   }

--- a/paddle/fluid/inference/api/api.cc
+++ b/paddle/fluid/inference/api/api.cc
@@ -92,7 +92,7 @@ void PaddleBuf::Reset(void *data, size_t length) {
 
 void PaddleBuf::Free() {
   if (memory_owned_ && data_) {
-    PADDLE_ENFORCE_GT(length_, 0);
+    PADDLE_ENFORCE_GT(length_, 0UL);
     free(static_cast<char *>(data_));
     data_ = nullptr;
     length_ = 0;

--- a/paddle/fluid/inference/tests/api/analyzer_seq_pool1_tester.cc
+++ b/paddle/fluid/inference/tests/api/analyzer_seq_pool1_tester.cc
@@ -56,14 +56,14 @@ struct DataRecord {
       std::vector<float> slot_data;
       split_to_float(data[1], ' ', &slot_data);
       std::string name = data[0];
-      PADDLE_ENFORCE_EQ(slot_data.size() % 11, 0,
+      PADDLE_ENFORCE_EQ(slot_data.size() % 11, 0UL,
                         "line %d, %s should be divisible", num_lines, name);
       datasets[name].emplace_back(std::move(slot_data));
     }
     num_samples = num_lines / num_slots;
     PADDLE_ENFORCE_EQ(num_samples * num_slots, static_cast<size_t>(num_lines),
                       "num samples should be divisible");
-    PADDLE_ENFORCE_GT(num_samples, 0);
+    PADDLE_ENFORCE_GT(num_samples, 0UL);
   }
 
   void Prepare(int bs) {

--- a/paddle/fluid/operators/attention_lstm_op.cc
+++ b/paddle/fluid/operators/attention_lstm_op.cc
@@ -293,7 +293,7 @@ class AttentionLSTMKernel : public framework::OpKernel<T> {
       int len = x_lod[0][i + 1] - x_lod[0][i];
       max_seq_len = max_seq_len < len ? len : max_seq_len;
     }
-    PADDLE_ENFORCE_EQ(x_lod.size(), 1, "Input(X)'s lod size must be 1.");
+    PADDLE_ENFORCE_EQ(x_lod.size(), 1UL, "Input(X)'s lod size must be 1.");
     PADDLE_ENFORCE_EQ(c0->dims()[0], N, "C0 dims should be %d x %d.", N, D);
     fc_out->Resize({max_seq_len, 1});
 

--- a/paddle/fluid/operators/controlflow/get_places_op.cc
+++ b/paddle/fluid/operators/controlflow/get_places_op.cc
@@ -52,7 +52,7 @@ class GetPlacesOp : public framework::OperatorBase {
       device_count =
           is_gpu ? CUDADevCount() : std::thread::hardware_concurrency();
     }
-    PADDLE_ENFORCE_NE(device_count, 0, "Cannot indicate %s device count",
+    PADDLE_ENFORCE_NE(device_count, 0UL, "Cannot indicate %s device count",
                       is_gpu ? "GPU" : "CPU");
 
     auto out_var_name = Output("Out");

--- a/paddle/fluid/operators/crf_decoding_op.cc
+++ b/paddle/fluid/operators/crf_decoding_op.cc
@@ -84,12 +84,12 @@ class CRFDecodingOp : public framework::OperatorWithKernel {
                    "Output(ViterbiPath) should be not null.");
 
     auto emission_dims = ctx->GetInputDim("Emission");
-    PADDLE_ENFORCE_EQ(emission_dims.size(), 2UL,
+    PADDLE_ENFORCE_EQ(emission_dims.size(), 2,
                       "The Input(Emission) should be a 2-D tensor.");
     PADDLE_ENFORCE(emission_dims[0], "An empty mini-batch is not allowed.");
 
     auto transition_dims = ctx->GetInputDim("Transition");
-    PADDLE_ENFORCE_EQ(transition_dims.size(), 2UL,
+    PADDLE_ENFORCE_EQ(transition_dims.size(), 2,
                       "The Input(Transition) should be a 2-D tensor.");
     PADDLE_ENFORCE_EQ(
         transition_dims[0] - 2, transition_dims[1],

--- a/paddle/fluid/operators/detection/anchor_generator_op.cc
+++ b/paddle/fluid/operators/detection/anchor_generator_op.cc
@@ -85,7 +85,7 @@ class AnchorGeneratorOpMaker : public framework::OpProtoAndCheckerMaker {
         " For instance, the anchor size of 64 means the area of this anchor "
         "equals to 64**2.")
         .AddCustomChecker([](const std::vector<float>& anchor_sizes) {
-          PADDLE_ENFORCE_GT(anchor_sizes.size(), 0,
+          PADDLE_ENFORCE_GT(anchor_sizes.size(), 0UL,
                             "Size of anchor_sizes must be at least 1.");
           for (size_t i = 0; i < anchor_sizes.size(); ++i) {
             PADDLE_ENFORCE_GT(anchor_sizes[i], 0.0,
@@ -103,7 +103,7 @@ class AnchorGeneratorOpMaker : public framework::OpProtoAndCheckerMaker {
                                 "(vector<float>) List of variances to be used "
                                 "in box regression deltas")
         .AddCustomChecker([](const std::vector<float>& variances) {
-          PADDLE_ENFORCE_EQ(variances.size(), 4,
+          PADDLE_ENFORCE_EQ(variances.size(), 4UL,
                             "Must and only provide 4 variance.");
           for (size_t i = 0; i < variances.size(); ++i) {
             PADDLE_ENFORCE_GT(variances[i], 0.0,
@@ -117,7 +117,7 @@ class AnchorGeneratorOpMaker : public framework::OpProtoAndCheckerMaker {
         .SetDefault(std::vector<float>(2, 16.0))
         .AddCustomChecker([](const std::vector<float>& stride) {
           PADDLE_ENFORCE_EQ(
-              stride.size(), 2,
+              stride.size(), 2UL,
               "Must and only provide 2 stride for width and height.");
           for (size_t i = 0; i < stride.size(); ++i) {
             PADDLE_ENFORCE_GT(stride[i], 0.0,

--- a/paddle/fluid/operators/fc_op.cc
+++ b/paddle/fluid/operators/fc_op.cc
@@ -47,7 +47,7 @@ void FCOp::InferShape(framework::InferShapeContext* ctx) const {
     PADDLE_ENFORCE(in_dims.size() == 2 || in_dims.size() == 4,
                    "Fully Connected input should be 2-D or 4-D tensor.");
   }
-  PADDLE_ENFORCE_EQ(w_dims.size(), 2UL,
+  PADDLE_ENFORCE_EQ(w_dims.size(), 2,
                     "Fully Connected input should be 2-D tensor.");
   int in_num_col_dims = ctx->Attrs().Get<int>("in_num_col_dims");
   PADDLE_ENFORCE_GT(

--- a/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h
+++ b/paddle/fluid/operators/fused/fused_embedding_seq_pool_op.h
@@ -47,10 +47,11 @@ struct EmbeddingVSumFunctor {
     auto *output = output_t->mutable_data<T>(context.GetPlace());
 
     PADDLE_ENFORCE_LE(table_width * idx_width, out_width);
+    PADDLE_ENFORCE_GT(ids_lod.size(), 1UL);
 
     jit::emb_seq_pool_attr_t attr(table_height, table_width, 0, idx_width,
                                   out_width, jit::SeqPoolType::kSum);
-    for (int64_t i = 0; i != ids_lod.size() - 1; ++i) {
+    for (size_t i = 0; i != ids_lod.size() - 1; ++i) {
       attr.index_height = ids_lod[i + 1] - ids_lod[i];
       auto emb_seqpool = jit::Get<jit::kEmbSeqPool, jit::EmbSeqPoolTuples<T>,
                                   platform::CPUPlace>(attr);

--- a/paddle/fluid/operators/fused/fusion_repeated_fc_relu_op.cc
+++ b/paddle/fluid/operators/fused/fusion_repeated_fc_relu_op.cc
@@ -37,7 +37,7 @@ void FusionRepeatedFCReluOp::InferShape(
                  "Output(Out) of FusionRepeatedFCReluOp should not be null.");
 
   auto i_dims = ctx->GetInputDim("X");
-  PADDLE_ENFORCE_EQ(i_dims.size(), 2UL, "Input shape size should be 2");
+  PADDLE_ENFORCE_EQ(i_dims.size(), 2, "Input shape size should be 2");
 
   auto w_dims = ctx->GetInputsDim("W");
   auto b_dims = ctx->GetInputsDim("Bias");
@@ -49,7 +49,7 @@ void FusionRepeatedFCReluOp::InferShape(
                     "inpute width should be equal with weight height");
 
   for (size_t i = 1; i < sz; ++i) {
-    PADDLE_ENFORCE_EQ(w_dims[i].size(), 2UL,
+    PADDLE_ENFORCE_EQ(w_dims[i].size(), 2,
                       "Every weight shape size should be 2.");
     PADDLE_ENFORCE_EQ(framework::product(b_dims[i]), w_dims[i][1],
                       "The length of Bias must be equal with w_dims[1].");

--- a/paddle/fluid/operators/fused/fusion_seqexpand_concat_fc_op.cc
+++ b/paddle/fluid/operators/fused/fusion_seqexpand_concat_fc_op.cc
@@ -39,7 +39,7 @@ void FusionSeqExpandConcatFCOp::InferShape(
 
   auto ins_dims = ctx->GetInputsDim("X");
   auto w_dims = ctx->GetInputDim("FCWeight");  // (M0+M1+M2+..) x D
-  PADDLE_ENFORCE_EQ(w_dims.size(), 2UL, "Input(FCWeight)'s rank must be 2.");
+  PADDLE_ENFORCE_EQ(w_dims.size(), 2, "Input(FCWeight)'s rank must be 2.");
   const int D = w_dims[1];
   int sum = ins_dims[0][1];
   for (size_t i = 1; i < ins_dims.size(); ++i) {

--- a/paddle/fluid/operators/fused/fusion_seqpool_concat_op.cc
+++ b/paddle/fluid/operators/fused/fusion_seqpool_concat_op.cc
@@ -39,7 +39,7 @@ void FusionSeqPoolConcatOp::InferShape(
 
   // The output height should be confirmed in Compute,
   // since input lod is not accessible here.
-  PADDLE_ENFORCE_EQ(ins_dims[0].size(), 2UL,
+  PADDLE_ENFORCE_EQ(ins_dims[0].size(), 2,
                     "The dims size of first input should be 2.");
   ctx->SetOutputDim("Out", {-1, ins_dims[0][axis] * static_cast<int>(n)});
 }

--- a/paddle/fluid/operators/fused/fusion_squared_mat_sub_op.cc
+++ b/paddle/fluid/operators/fused/fusion_squared_mat_sub_op.cc
@@ -42,7 +42,7 @@ void FusionSquaredMatSubOp::InferShape(
   auto y_dims = ctx->GetInputDim("Y");
   PADDLE_ENFORCE_EQ(x_dims.size(), y_dims.size(),
                     "Input tensors dims size should be equal.");
-  PADDLE_ENFORCE_EQ(x_dims.size(), 2UL, "Input tensors should be a Matrix.");
+  PADDLE_ENFORCE_EQ(x_dims.size(), 2, "Input tensors should be a Matrix.");
   PADDLE_ENFORCE_EQ(x_dims[1], y_dims[0], "Inputs Matrix should be multiply.");
 
   ctx->SetOutputDim("SquaredX", x_dims);

--- a/paddle/fluid/operators/layer_norm_op.cc
+++ b/paddle/fluid/operators/layer_norm_op.cc
@@ -44,11 +44,11 @@ class LayerNormOp : public framework::OperatorWithKernel {
     int left = static_cast<int>(matrix_dim[0]);
     int right = static_cast<int>(matrix_dim[1]);
     if (ctx->HasInput("Scale")) {
-      PADDLE_ENFORCE_EQ(ctx->GetInputDim("Scale").size(), 1UL);
+      PADDLE_ENFORCE_EQ(ctx->GetInputDim("Scale").size(), 1);
       PADDLE_ENFORCE_EQ(ctx->GetInputDim("Scale")[0], right);
     }
     if (ctx->HasInput("Bias")) {
-      PADDLE_ENFORCE_EQ(ctx->GetInputDim("Bias").size(), 1UL);
+      PADDLE_ENFORCE_EQ(ctx->GetInputDim("Bias").size(), 1);
       PADDLE_ENFORCE_EQ(ctx->GetInputDim("Bias")[0], right);
     }
 

--- a/paddle/fluid/operators/linear_chain_crf_op.cc
+++ b/paddle/fluid/operators/linear_chain_crf_op.cc
@@ -144,12 +144,12 @@ class LinearChainCRFOp : public framework::OperatorWithKernel {
                    "Output(LogLikelihood) should be not null.");
 
     auto emission_dims = ctx->GetInputDim("Emission");
-    PADDLE_ENFORCE_EQ(emission_dims.size(), 2UL,
+    PADDLE_ENFORCE_EQ(emission_dims.size(), 2,
                       "The Input(Emission) should be a 2-D tensor.");
     PADDLE_ENFORCE(emission_dims[0], "An empty mini-batch is not allowed.");
 
     auto transition_dims = ctx->GetInputDim("Transition");
-    PADDLE_ENFORCE_EQ(transition_dims.size(), 2UL,
+    PADDLE_ENFORCE_EQ(transition_dims.size(), 2,
                       "The Input(Transition) should be a 2-D tensor.");
     PADDLE_ENFORCE_EQ(
         transition_dims[0] - 2, transition_dims[1],
@@ -202,13 +202,13 @@ class LinearChainCRFGradOp : public framework::OperatorWithKernel {
                    "Input(LogLikelihood@GRAD) shoudl be not null.");
 
     auto emission_exps_dims = ctx->GetInputDim("EmissionExps");
-    PADDLE_ENFORCE_EQ(emission_exps_dims.size(), 2UL,
+    PADDLE_ENFORCE_EQ(emission_exps_dims.size(), 2,
                       "The Input(EmissionExps) should be a 2-D tensor.");
     PADDLE_ENFORCE(emission_exps_dims[0],
                    "An empty mini-batch is not allowed.");
 
     auto transition_exps_dims = ctx->GetInputDim("TransitionExps");
-    PADDLE_ENFORCE_EQ(transition_exps_dims.size(), 2UL,
+    PADDLE_ENFORCE_EQ(transition_exps_dims.size(), 2,
                       "The Input(TransitionExps) should be a 2-D tensor.");
     PADDLE_ENFORCE_EQ(
         transition_exps_dims[0] - 2, transition_exps_dims[1],

--- a/paddle/fluid/operators/sequence_ops/sequence_enumerate_op.cc
+++ b/paddle/fluid/operators/sequence_ops/sequence_enumerate_op.cc
@@ -31,10 +31,10 @@ class SequenceEnumerateOp : public framework::OperatorWithKernel {
 
     const auto x_dims = ctx->GetInputDim("X");
     PADDLE_ENFORCE_EQ(
-        x_dims.size(), 2UL,
+        x_dims.size(), 2,
         "Input(X) of SequenceEnumerate operator's rank should be 2.");
     PADDLE_ENFORCE_EQ(
-        x_dims[1], 1UL,
+        x_dims[1], 1,
         "Input(X) of SequenceEnumerate operator's 2nd dimension should be 1.");
 
     const auto win_size = ctx->Attrs().Get<int>("win_size");

--- a/paddle/fluid/operators/sequence_ops/sequence_expand_op.cc
+++ b/paddle/fluid/operators/sequence_ops/sequence_expand_op.cc
@@ -48,10 +48,10 @@ class SequenceExpandOp : public framework::OperatorWithKernel {
       auto& x_lod = x_var->Get<LoDTensor>().lod();
       auto& y_lod = y_var->Get<LoDTensor>().lod();
 
-      PADDLE_ENFORCE_LE(x_lod.size(), 1,
+      PADDLE_ENFORCE_LE(x_lod.size(), 1UL,
                         "Level number of Input(X)'s lod should not be "
                         "greater than 1.");
-      PADDLE_ENFORCE_GT(y_lod.size(), 0,
+      PADDLE_ENFORCE_GT(y_lod.size(), 0UL,
                         "Level number of Input(Y)'s lod should be "
                         "greater than 0.");
       PADDLE_ENFORCE(
@@ -69,7 +69,8 @@ class SequenceExpandOp : public framework::OperatorWithKernel {
                        "size of Input(X)'s first level lod should be equal to "
                        "size of Input(Y)'s referred level lod.");
       } else {
-        PADDLE_ENFORCE_EQ(x_dims[0], y_lod[ref_level].size() - 1,
+        PADDLE_ENFORCE_EQ(x_dims[0],
+                          static_cast<int64_t>(y_lod[ref_level].size()) - 1,
                           "When Input(X)'s lod is null, the dims[0] of "
                           "Input(X) should match the "
                           "size of Input(Y)'s referred level lod.");

--- a/paddle/fluid/platform/enforce_test.cc
+++ b/paddle/fluid/platform/enforce_test.cc
@@ -121,9 +121,9 @@ TEST(ENFORCE_GT, FAIL) {
     PADDLE_ENFORCE_GT(1, 2);
   } catch (paddle::platform::EnforceNotMet error) {
     caught_exception = true;
-    EXPECT_TRUE(HasPrefix(
-        StringPiece(error.what()),
-        "Enforce failed. Expected 1 > 2UL, but received 1:1 <= 2UL:2."));
+    EXPECT_TRUE(
+        HasPrefix(StringPiece(error.what()),
+                  "Enforce failed. Expected 1 > 2, but received 1:1 <= 2:2."));
   }
   EXPECT_TRUE(caught_exception);
 }
@@ -139,9 +139,9 @@ TEST(ENFORCE_GE, FAIL) {
     PADDLE_ENFORCE_GE(1, 2);
   } catch (paddle::platform::EnforceNotMet error) {
     caught_exception = true;
-    EXPECT_TRUE(HasPrefix(
-        StringPiece(error.what()),
-        "Enforce failed. Expected 1 >= 2UL, but received 1:1 < 2UL:2."));
+    EXPECT_TRUE(
+        HasPrefix(StringPiece(error.what()),
+                  "Enforce failed. Expected 1 >= 2, but received 1:1 < 2:2."));
   }
   EXPECT_TRUE(caught_exception);
 }
@@ -156,12 +156,12 @@ TEST(ENFORCE_LE, OK) {
 TEST(ENFORCE_LE, FAIL) {
   bool caught_exception = false;
   try {
-    PADDLE_ENFORCE_GT(1, 2UL);
+    PADDLE_ENFORCE_GT(1, 2);
   } catch (paddle::platform::EnforceNotMet error) {
     caught_exception = true;
-    EXPECT_TRUE(HasPrefix(
-        StringPiece(error.what()),
-        "Enforce failed. Expected 1 > 2UL, but received 1:1 <= 2UL:2."));
+    EXPECT_TRUE(
+        HasPrefix(StringPiece(error.what()),
+                  "Enforce failed. Expected 1 > 2, but received 1:1 <= 2:2."));
   }
   EXPECT_TRUE(caught_exception);
 }

--- a/paddle/fluid/platform/enforce_test.cc
+++ b/paddle/fluid/platform/enforce_test.cc
@@ -118,7 +118,7 @@ TEST(ENFORCE_GT, OK) { PADDLE_ENFORCE_GT(2, 1); }
 TEST(ENFORCE_GT, FAIL) {
   bool caught_exception = false;
   try {
-    PADDLE_ENFORCE_GT(1, 2UL);
+    PADDLE_ENFORCE_GT(1, 2);
   } catch (paddle::platform::EnforceNotMet error) {
     caught_exception = true;
     EXPECT_TRUE(HasPrefix(
@@ -129,15 +129,14 @@ TEST(ENFORCE_GT, FAIL) {
 }
 
 TEST(ENFORCE_GE, OK) {
-  PADDLE_ENFORCE_GE(2, 2UL);
-  PADDLE_ENFORCE_GE(3, 2UL);
+  PADDLE_ENFORCE_GE(2, 2);
   PADDLE_ENFORCE_GE(3, 2);
-  PADDLE_ENFORCE_GE(3.21, 2UL);
+  PADDLE_ENFORCE_GE(3.21, 2.0);
 }
 TEST(ENFORCE_GE, FAIL) {
   bool caught_exception = false;
   try {
-    PADDLE_ENFORCE_GE(1, 2UL);
+    PADDLE_ENFORCE_GE(1, 2);
   } catch (paddle::platform::EnforceNotMet error) {
     caught_exception = true;
     EXPECT_TRUE(HasPrefix(
@@ -149,10 +148,10 @@ TEST(ENFORCE_GE, FAIL) {
 
 TEST(ENFORCE_LE, OK) {
   PADDLE_ENFORCE_LE(1, 1);
-  PADDLE_ENFORCE_LE(1, 1UL);
-  PADDLE_ENFORCE_LE(2, 3UL);
-  PADDLE_ENFORCE_LE(2UL, 3);
-  PADDLE_ENFORCE_LE(2UL, 3.2);
+  PADDLE_ENFORCE_LE(1UL, 1UL);
+  PADDLE_ENFORCE_LE(2, 3);
+  PADDLE_ENFORCE_LE(2UL, 3UL);
+  PADDLE_ENFORCE_LE(2.0, 3.2);
 }
 TEST(ENFORCE_LE, FAIL) {
   bool caught_exception = false;
@@ -169,8 +168,8 @@ TEST(ENFORCE_LE, FAIL) {
 
 TEST(ENFORCE_LT, OK) {
   PADDLE_ENFORCE_LT(3, 10);
-  PADDLE_ENFORCE_LT(2, 3UL);
-  PADDLE_ENFORCE_LT(2UL, 3);
+  PADDLE_ENFORCE_LT(2UL, 3UL);
+  PADDLE_ENFORCE_LT(2, 3);
 }
 TEST(ENFORCE_LT, FAIL) {
   bool caught_exception = false;

